### PR TITLE
Remove publication_year and interpolation meta

### PR DESCRIPTION
**Pull request by YosuaMichael**
resolving https://github.com/pytorch/vision/issues/5768

### CODE REVIEW

File: torchvision/models/vision_transformer.py
Comment (YosuaMichael) [inline torchvision/models/vision_transformer.py:326]: Add the type explicitly to avoid mypy triggering error

File: torchvision/models/vision_transformer.py
Comment (datumbox) [inline torchvision/models/vision_transformer.py:321]: I assume mypy is to blame for this, right?

**Review by datumbox — APPROVED** [review]
LGTM, thanks @YosuaMichael!

File: torchvision/models/vision_transformer.py
Comment (YosuaMichael) [reply torchvision/models/vision_transformer.py:321]: Yes, correct

**Comment by datumbox** [comment]
The failures seem unrelated. I've raised the issues separately at #5850 and #5830

**Comment by YosuaMichael** [comment]
> The failures seem unrelated. I've raised the issues separately at #5850 and #5830

I will wait until all the tests finished first, just to be safe. After that I will merge if no other test that break.

**Pull request merged**
